lark/parsers/lalr_parser.py

```diff
@@ -35,84 +35,116 @@ def parse(self, *args):
         return self.parser.parse(*args)
 
 
-class _Parser:
-    def __init__(self, parse_table, callbacks, debug=False):
-        self.parse_table = parse_table
-        self.callbacks = callbacks
-        self.debug = debug
+class ParserState:
+    __slots__ = 'parse_table', 'lexer', 'callbacks', 'start', 'state_stack', 'value_stack', 'start_state', 'end_state', 'states'
```

Comment:
Only `lexer_thread`, `value_stack` and `state_stack` actually represent the State. I think the rest should be accesed via a `parser` attribute.

Reply:
I save them in `ParserState` for performance reasons, otherwise the `feed_token()` method would have to fetch them again and again in every iteration.

I can prefix them with `_` if you think that would make it clearer.

Reply:
Did you measure it? I think it doesn't matter whether or not you get the attribute twice or just once. (e.g. `self.lexer.parse_table` vs `self.parse_table`)

Reply:
I did measure it actually, but I didn't try to put `__slots__` in `ParseTable`. I guess it's worth a try.